detection_rules/kbwrap.py

```diff
@@ -107,35 +112,55 @@ def kibana_import_rules(ctx: click.Context, rules: RuleCollection, overwrite: Op
     return response, results
 
 
-@kibana_group.command('export-rules')
-@click.option('--directory', '-d', required=True, type=Path, help='Directory to export rules to')
-@click.option('--rule-id', '-r', multiple=True, help='Optional Rule IDs to restrict export to')
-@click.option('--skip-errors', '-s', is_flag=True, help='Skip errors when exporting rules')
-@click.option('--strip-version', '-sv', is_flag=True, help='Strip the version fields from all rules')
+@kibana_group.command("export-rules")
+@click.option("--directory", "-d", required=True, type=Path, help="Directory to export rules to")
+@click.option("--exceptions-directory", "-ed", required=False, type=Path, help="Directory to export exceptions to")
+@click.option("--rule-id", "-r", multiple=True, help="Optional Rule IDs to restrict export to")
+@click.option("--export-exceptions", "-e", is_flag=True, help="Include exceptions in export")
+@click.option("--skip-errors", "-s", is_flag=True, help="Skip errors when exporting rules")
+@click.option("--strip-version", "-sv", is_flag=True, help="Strip the version fields from all rules")
 @click.pass_context
-def kibana_export_rules(ctx: click.Context, directory: Path, rule_id: Optional[Iterable[str]] = None,
-                        skip_errors: bool = False, strip_version: bool = False) -> List[TOMLRule]:
+def kibana_export_rules(
+    ctx: click.Context,
+    directory: Path,
+    exceptions_directory: Optional[Path],
+    rule_id: Optional[Iterable[str]] = None,
+    export_exceptions: bool = False,
+    skip_errors: bool = False,
+    strip_version: bool = False,
+) -> List[TOMLRule]:
     """Export custom rules from Kibana."""
-    kibana = ctx.obj['kibana']
+    kibana = ctx.obj["kibana"]
     with kibana:
-        results = RuleResource.export_rules(list(rule_id))
+        results = RuleResource.export_rules(list(rule_id), exclude_export_details=not export_exceptions)
 
     if results:
         directory.mkdir(parents=True, exist_ok=True)
 
+    rules_results = results
+    if export_exceptions:
+        # Assign counts to variables
+        rules_count = results[-1]["exported_rules_count"]
+        exception_list_count = results[-1]["exported_exception_list_count"]
+        exception_list_item_count = results[-1]["exported_exception_list_item_count"]
```

Comment:
if there is a chance there are no results, the execution will arrive here and it will break

Reply:
Good call! Will update this  :+1: